### PR TITLE
RR-501 Add a new goal - Change step description

### PR DESCRIPTION
**Pull request by chrimesdev**
## Description 

Add a new goal - Change step description

https://dsdmoj.atlassian.net/browse/RR-501

## Screenshots/Video

https://github.com/ministryofjustice/hmpps-education-and-work-plan-ui/assets/11615501/0a7a58bd-56ae-4c23-8216-b315d818c170

### CODE REVIEW

File: server/views/pages/goal/review/index.njk [hunk 1]
```diff
@@ -60,6 +62,13 @@
                 <dd class="govuk-summary-list__value">
                   {{ step.title }}
                 </dd>
+                {% if featureToggles.newCreateGoalRoutesEnabled %}
+                <dd class="govuk-summary-list__actions">
+                  <a class="govuk-link" href="/plan/{{ prisonerSummary.prisonNumber }}/goals/{{ loop.index }}/add-step/{{ step.sequenceNumber }}?mode=edit" data-q="change-goal-{{ loop.index }}-step-{{ step.sequenceNumber }}">
```
Comment (chrimesdev) [inline server/views/pages/goal/review/index.njk:67]: Need to figure out how to access to the goal number here, `{{ loop.index }}` won't work, as we're in a separate loop for the "Steps" and Goal doesn't have a `sequenceNumber` unlike Steps.

Comment (chrimesdev) [reply server/views/pages/goal/review/index.njk]: `{% set goalLoop = loop %}`

File: server/views/pages/goal/review/index.njk [hunk 2]
```diff
@@ -16,23 +16,24 @@
       <h1 class="govuk-heading-l">Review {{ prisonerSummary.firstName }} {{ prisonerSummary.lastName }}'s goals</h1>
 
       {% for goal in goals %}
+      {% set goalLoop = loop %}
```
Comment (chrimesdev) [inline server/views/pages/goal/review/index.njk:19]: This is so we can track the `{% for goal in goals %}` index number for Change link URL nested within the `{% for step in goal.steps %}` loop (see above comment)

File: integration_tests/e2e/goal/addAndEditStep.cy.ts [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+      .setStepTitle('Book French course')
+      .submitPage()
+
+    const addNotePage = Page.verifyOnPage(AddNotePage)
+    addNotePage.setNote("Pay close attention to Chris' behaviour during classes")
+    addNotePage.submitPage()
+
+    // When
+    const reviewPage = Page.verifyOnPage(ReviewPage)
+    reviewPage.clickChangeGoalDescriptionLink()
```
[the remark this answers is not in the document]
Comment (chrimesdev) [reply integration_tests/e2e/goal/addAndEditStep.cy.ts]: Good spot - fixed

**Review by nathanrussell-moj-digital — APPROVED** [review]
LGTM 👍

**Pull request merged**